src/pudl/transform/eia861.py

```diff
@@ -1248,19 +1220,19 @@ def demand_response(tfr_dfs):
         "report_date",
     ]
 
-    raw_dr = tfr_dfs["demand_response_eia861"].copy()
+    raw_dr = _pre_process(raw_demand_response_eia861)
     # fill na BA values with 'UNK'
     raw_dr["balancing_authority_code_eia"] = raw_dr[
         "balancing_authority_code_eia"
     ].fillna("UNK")
     raw_dr["short_form"] = _make_yn_bool(raw_dr.short_form)
 
     # Split data into tidy-able and not
-    raw_dr_water_heater = raw_dr[idx_cols + ["water_heater"]].copy()
-    raw_dr_water_heater = _drop_dupes(
+    raw_dr_water_heater = raw_dr[idx_cols + ["water_heater", "data_maturity"]].copy()
+    dr_water = _drop_dupes(
```

Comment:
A lot of the little futzing around in all these functions is to ensure that `data_maturity` propagates to all of the outputs.